src/main/java/tr/com/srdc/cda2fhir/jolt/AdditionalModifier.java

```diff
@@ -205,6 +207,113 @@ protected Optional applyList(final List<Object> argList) {
 		}
 	}
 
+	private static String iterableJoin(Iterable<String> iterable, String delimiter) {
+		String joined = "";
+
+		Iterator<String> iterator = iterable.iterator();
+
+		while (iterator.hasNext()) {
+			joined += iterator.next();
+			if (iterator.hasNext())
+				joined += delimiter;
+		}
+
+		return joined;
+
+	}
+
+	@SuppressWarnings("unchecked")
+	public static final class GetReferenceDisplay extends Function.SingleFunction<Object> {
+		@Override
+		protected Optional<Object> applySingle(final Object arg) {
```

Comment:
Good for now. I understand your comment from last week that majority of this "path" finding should be in jolt proper. I think I already had something for this; a second shift transform. I will take a look after this is in.

Reply:
Awesome, glad you get what I meant.